<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Multivariable MR Tutorial • MVMR</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Multivariable MR Tutorial">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MVMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/MVMR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/WSpiller/MVMR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Multivariable MR Tutorial</h1>
                        <h4 data-toc-skip class="author">Wes Spiller,
Eleanor Sanderson, and Jack Bowden</h4>
            
            <h4 data-toc-skip class="date">31<sup>st</sup> January
2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/WSpiller/MVMR/blob/HEAD/vignettes/MVMR.rmd" class="external-link"><code>vignettes/MVMR.rmd</code></a></small>
      <div class="d-none name"><code>MVMR.rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Multivariable Mendelian Randomisation (MVMR) is a form of
instrumental variable analysis which estimates the direct effect of
multiple exposures on an outcome using genetic variants as instruments.
The <code>MVMR</code> R package facilitates estimation of causal effects
using MVMR, as well as including a range of sensitivity analyses
evaluating the underlying assumptions of the approach. The methods
included in <code>MVMR</code> originate from <span class="citation">Sanderson, Spiller, and Bowden (2021)</span>, available
<a href="https://doi.org/10.1002/sim.9133" class="external-link">here</a>.</p>
<div class="section level3">
<h3 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h3>
<p>Fitting and interpreting MVMR models can be achieved by following the
5 steps given below:</p>
<ol style="list-style-type: decimal">
<li>Obtain data</li>
<li>Format data</li>
<li>Assess instrument strength</li>
<li>Assess horizontal pleiotropy</li>
<li>Estimate causal effects</li>
</ol>
<p>Each of these steps are shown in the figure below, highlighting the R
function used for each step.</p>
<div class="figure" style="text-align: center">
<img src="png/workflow.png" alt="Figure showing the workflow of an MVMR analysis." width="600px"><p class="caption">
The workflow of an MVMR analysis.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="step-1-obtain-summary-data">Step 1: Obtain summary data<a class="anchor" aria-label="anchor" href="#step-1-obtain-summary-data"></a>
</h2>
<p>The following information is necessary to estimate causal effects
using MVMR:</p>
<ol style="list-style-type: decimal">
<li><p>Gene-exposure associations for each variant selected as an
instrument for any exposure.</p></li>
<li><p>Corresponding standard errors for the gene-exposure
associations.</p></li>
<li><p>Gene-outcome associations for each instrument.</p></li>
<li><p>Corresponding standard errors for the gene-outcome
associations.</p></li>
</ol>
<p>The data frame <code>rawdat_mvmr</code>, included in the
<code>MVMR</code> package shows an example of such data obtained from
MRBase. When data is extracted from MRBase or using the TwoSampleMR R
package, the <code><a href="../reference/mrmvinput_to_mvmr_format.html">mrmvinput_to_mvmr_format()</a></code> function can be
used to convert the data to the <code>MVMR</code> data format. This
specifically requires gene-exposure associations for all included SNPs,
gene-outcome associations, and corresponding standard errors. In this
case, low-density lipoprotein cholesterol (LDL-C), high-density
lipoprotein cholesterol (HDL-C), and triglycerides (Trg) have been
selected as exposures, while systolic blood pressure (SBP) is the
outcome of interest. Here the suffix <code>_beta</code> is used to
denote association estimates, while <code>_se</code> denotes standard
errors. Please note that the <code>MVMR</code> can take an arbirtary
number of exposures (greater than 1), and that three exposures have been
selected purely for illustration.</p>
<p>The first 6 rows of <code>rawdat_mvmr</code> are:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/WSpiller/MVMR" class="external-link">MVMR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rawdat_mvmr</span><span class="op">)</span></span>
<span><span class="co">#&gt;   LDL_beta HDL_beta Trg_beta LDL_se HDL_se Trg_se    SBP_beta     SBP_se</span></span>
<span><span class="co">#&gt; 1  -0.0270   0.0182   0.0228 0.0046 0.0050 0.0045 -0.00426935 0.00280123</span></span>
<span><span class="co">#&gt; 2   0.1179   0.0020   0.0379 0.0038 0.0042 0.0039  0.00110389 0.00227690</span></span>
<span><span class="co">#&gt; 3  -0.0269   0.0079   0.0000 0.0039 0.0042 0.0038 -0.01317370 0.00222743</span></span>
<span><span class="co">#&gt; 4   0.0081  -0.0508   0.0085 0.0064 0.0069 0.0062 -0.00111303 0.00374264</span></span>
<span><span class="co">#&gt; 5   0.0191   0.0103  -0.0270 0.0034 0.0036 0.0033 -0.00854986 0.00207701</span></span>
<span><span class="co">#&gt; 6   0.0043   0.0320   0.0109 0.0038 0.0040 0.0037  0.00472509 0.00237133</span></span>
<span><span class="co">#&gt;          SNP</span></span>
<span><span class="co">#&gt; 1 rs10019888</span></span>
<span><span class="co">#&gt; 2 rs10468017</span></span>
<span><span class="co">#&gt; 3  rs1047891</span></span>
<span><span class="co">#&gt; 4 rs10490626</span></span>
<span><span class="co">#&gt; 5 rs10761762</span></span>
<span><span class="co">#&gt; 6 rs10832962</span></span></code></pre></div>
<p>Note that the final column <code>SNP</code> contains the rsid numbers
for each genetic variant. These are not necessary for conducting MVMR,
but assist in follow-up analyses. Summary data for LDL-C, HDL-C, and
Triglycerides originate from <a href="https://www.nature.com/articles/ng.2797" class="external-link">GLGC</a>, while SBP data
was obtained using <a href="https://www.nature.com/articles/ng.3768" class="external-link">UK
Biobank</a>.</p>
<div class="section level3">
<h3 id="estimating-pairwise-covariances-between-snp-associations">Estimating pairwise covariances between SNP associations<a class="anchor" aria-label="anchor" href="#estimating-pairwise-covariances-between-snp-associations"></a>
</h3>
<p>The MVMR approach requires pairwise covariances between an instrument
and pairs of exposures to be known across all SNPs for testing and
sensitivity analyses. However, this is often not reported in published
GWAS analyses. Before continuing with MVMR it is therefore
<strong>necessary</strong> to select one of the following three
solutions:</p>
<ol style="list-style-type: decimal">
<li>
<p>Estimate the covariance terms using individual level data</p>
<p>If individual level data is available from which the GWAS summary
estimates were obtained, the <code><a href="../reference/snpcov_mvmr.html">snpcov_mvmr()</a></code> function can be
used to calculate the necessary covariance terms.</p>
</li>
<li><p>Estimate the covariance terms using phenotypic correlation
between exposures. an estimate of the correlation between the
(phenotypic) exposures is available, the <code><a href="../reference/phenocov_mvmr.html">phenocov_mvmr()</a></code>
function can be used to provide an approximation for the necessary
covariance terms. This function takes the phenotypic correlation between
the exposures and the standard error of the SNP-exposure betas as
inputs.</p></li>
<li>
<p>Obtain gene-exposure associations from non-overlapping
samples.</p>
<p>If gene-exposure associations are estimated in seperate
non-overlapping samples, then the covariances will be zero by design. It
is therefore not necessary to calculate the set of covariances, although
this approach can be difficult to apply due to a lack of suitable
sources of data.</p>
</li>
</ol>
<p>When the necessary data are provided to the
<code><a href="../reference/snpcov_mvmr.html">snpcov_mvmr()</a></code> or <code><a href="../reference/phenocov_mvmr.html">phenocov_mvmr()</a></code> functions, a
set of covariance matrices will be produced equal to the number of SNPs
used in estimation. By saving this output as an object, it is possible
to use this information in downstream senstivity analyses and assumption
testing. As the <code><a href="../reference/phenocov_mvmr.html">phenocov_mvmr()</a></code> function requires
gene-exposure standard errors, it can be useful to estimate the
covariance matrices after initially formatiing the data. An illustrative
example is provided in step 6, creating an object <code>Xcovmat</code>
using the <code><a href="../reference/phenocov_mvmr.html">phenocov_mvmr()</a></code> function.</p>
</div>
</div>
<div class="section level2">
<h2 id="step-2-format-summary-data">Step 2: Format summary data<a class="anchor" aria-label="anchor" href="#step-2-format-summary-data"></a>
</h2>
<p>Downstream functions in the <code>MVMR</code> package rely upon prior
formatting of raw summary data using the <code><a href="../reference/format_mvmr.html">format_mvmr()</a></code>
function. Specifically, <code><a href="../reference/format_mvmr.html">format_mvmr()</a></code> checks and organises
summary data columns for use in MVMR analyses. The
<code><a href="../reference/format_mvmr.html">format_mvmr()</a></code> function takes the following arguments:</p>
<ul>
<li>
<code>BXGs</code>: A subset containing beta-coefficient values for
genetic associations with each exposure. Columns should indicate
exposure number, with rows representing estimates for a given genetic
variant.</li>
<li>
<code>BYG</code>: A numeric vector of beta-coefficient values for
genetic associations with the outcome.</li>
<li>
<code>seBXGs</code>: A subset containing standard errors
corresponding to the subset of beta-coefficients <code>BXGs</code>.</li>
<li>
<code>seBYG</code>: A numeric vector of standard errors
corresponding to the beta-coefficients <code>BYG</code>.</li>
<li>
<code>RSID</code>: A vector of names for genetic variants included
in the analysis. If variant IDs are not provided
(<code>RSID = "NULL"</code>), a vector of ID numbers will be
generated.</li>
</ul>
<p>Using the previous data <code>rawdat.mvmr</code>, we can format the
data using the following command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">F.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_mvmr.html">format_mvmr</a></span><span class="op">(</span></span>
<span>  BXGs <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  BYG <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">[</span>, <span class="fl">7</span><span class="op">]</span>,</span>
<span>  seBXGs <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  seBYG <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">[</span>, <span class="fl">8</span><span class="op">]</span>,</span>
<span>  RSID <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">[</span>, <span class="fl">9</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">F.data</span><span class="op">)</span></span>
<span><span class="co">#&gt;          SNP      betaYG   sebetaYG  betaX1  betaX2  betaX3 sebetaX1 sebetaX2</span></span>
<span><span class="co">#&gt; 1 rs10019888 -0.00426935 0.00280123 -0.0270  0.0182  0.0228   0.0046   0.0050</span></span>
<span><span class="co">#&gt; 2 rs10468017  0.00110389 0.00227690  0.1179  0.0020  0.0379   0.0038   0.0042</span></span>
<span><span class="co">#&gt; 3  rs1047891 -0.01317370 0.00222743 -0.0269  0.0079  0.0000   0.0039   0.0042</span></span>
<span><span class="co">#&gt; 4 rs10490626 -0.00111303 0.00374264  0.0081 -0.0508  0.0085   0.0064   0.0069</span></span>
<span><span class="co">#&gt; 5 rs10761762 -0.00854986 0.00207701  0.0191  0.0103 -0.0270   0.0034   0.0036</span></span>
<span><span class="co">#&gt; 6 rs10832962  0.00472509 0.00237133  0.0043  0.0320  0.0109   0.0038   0.0040</span></span>
<span><span class="co">#&gt;   sebetaX3</span></span>
<span><span class="co">#&gt; 1   0.0045</span></span>
<span><span class="co">#&gt; 2   0.0039</span></span>
<span><span class="co">#&gt; 3   0.0038</span></span>
<span><span class="co">#&gt; 4   0.0062</span></span>
<span><span class="co">#&gt; 5   0.0033</span></span>
<span><span class="co">#&gt; 6   0.0037</span></span></code></pre></div>
<p>In the above code we have provided the numbered columns for each
argument. For example, <code>BXGs = rawdat.mvmr[, c(1, 2, 3)]</code>
indicates that columns 1, 2, and 3 are the association estimates for
exposures 1, 2, and 3. It is important to note that standard error
columns <code>seBXGs</code> should be input in the same order as BXGs to
ensure the correct matching of association estimates with corresponding
standard errors.</p>
<p>In subsequent steps, each exposure is numbered such that
<code>X1</code>, <code>X2</code>, and <code>X3</code> are the first,
second, and third entries in the
<code>BXGs = rawdat.mvmr[, c(1, 2, 3)]</code> argument.</p>
</div>
<div class="section level2">
<h2 id="step-3-test-for-weak-instruments">Step 3: Test for weak instruments<a class="anchor" aria-label="anchor" href="#step-3-test-for-weak-instruments"></a>
</h2>
<p>In univariate two-sample summary MR, genetic variants selected as
instruments are required to be strongly associated with their
corresponding exposure. This is quantified by regressing the exposure
upon each instrument, and evaluating conditional dependence using the
F-statistic for the instrument. Conventionally, a F-statistic greater
than 10 is used as a threshold for sufficient instrument strength,
representing a 10% relative bias towards the null in the two-sample MR
setting.</p>
<p>Multivariable MR relies upon an extension of this assumption,
requiring instruments to be strongly associated with their corresponding
exposure conditioning on the remaining included exposures. Conditional
instrument strength is quantified by a conditional F-statistic which has
the same distribution as the univariate F-statistic. Consequently, the
same conventional instrument strength threshold of 10 can be used.</p>
<p>Further details are available <a href="https://doi.org/10.1002/sim.9133" class="external-link">here</a>.</p>
<p>The <code><a href="../reference/strength_mvmr.html">strength_mvmr()</a></code> function is used to evaluate
instrument strength in the MVMR setting. The function contains two
arguments:</p>
<ul>
<li>
<code>r_input</code>: A formatted data frame created using the
<code><a href="../reference/format_mvmr.html">format_mvmr()</a></code> function or an object of class
<code>MRMVInput</code> from the <code><a href="https://rdrr.io/pkg/MendelianRandomization/man/mr_mvinput.html" class="external-link">mr_mvinput()</a></code> function in
the MendelianRandomization package.</li>
<li>
<code>gencov</code>: A variance-covariance matrix for the effect of
the genetic variants on each exposure. This is obtained from either
<code><a href="../reference/snpcov_mvmr.html">snpcov_mvmr()</a></code>, <code><a href="../reference/phenocov_mvmr.html">phenocov_mvmr()</a></code>, or set to zero
when omitted.</li>
</ul>
<p><strong>Note</strong>: The <code><a href="../reference/strength_mvmr.html">strength_mvmr()</a></code> function will
output a warning if a variance-covariance matrix is not provided. Please
see Step 1 for further information.</p>
<p>Continuing with the previous example, we can evaluate the conditional
strength of the instruments for each exposure using the following
command</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sres</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/strength_mvmr.html">strength_mvmr</a></span><span class="op">(</span>r_input <span class="op">=</span> <span class="va">F.data</span>, gencov <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in strength_mvmr(r_input = F.data, gencov = 0): Covariance between</span></span>
<span><span class="co">#&gt; effect of genetic variants on each exposure not specified. Fixing covariance at</span></span>
<span><span class="co">#&gt; 0.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Conditional F-statistics for instrument strength</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             exposure1 exposure2 exposure3</span></span>
<span><span class="co">#&gt; F-statistic  46.33671  67.80463  38.80184</span></span></code></pre></div>
<p>In this case the set of instruments is sufficiently strong for MVMR
estimation using the conventional F-statistic threshold of 10. However,
note that we have manually set <code>gencov</code> to zero, which would
likely not be appropriate given each SNP-exposure estimate was obtained
from the same sample. Using a random phenotypic correlation matrix,
conditional F-statistics can be calculated as</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mvmrcormatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">0.1</span>, <span class="op">-</span><span class="fl">0.05</span>, <span class="op">-</span><span class="fl">0.1</span>, <span class="fl">1</span>, <span class="fl">0.2</span>, <span class="op">-</span><span class="fl">0.05</span>, <span class="fl">0.2</span>, <span class="fl">1</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">Xcovmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phenocov_mvmr.html">phenocov_mvmr</a></span><span class="op">(</span><span class="va">mvmrcormatrix</span>, <span class="va">F.data</span><span class="op">[</span>, <span class="fl">7</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">sres2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/strength_mvmr.html">strength_mvmr</a></span><span class="op">(</span>r_input <span class="op">=</span> <span class="va">F.data</span>, gencov <span class="op">=</span> <span class="va">Xcovmat</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Conditional F-statistics for instrument strength</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             exposure1 exposure2 exposure3</span></span>
<span><span class="co">#&gt; F-statistic  48.20993  69.55193  39.77326</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-4-test-for-horizontal-pleiotropy-using-conventional-q-statistic-estimation">Step 4: Test for horizontal pleiotropy using conventional
Q-statistic estimation<a class="anchor" aria-label="anchor" href="#step-4-test-for-horizontal-pleiotropy-using-conventional-q-statistic-estimation"></a>
</h2>
<p>Horizontal pleiotropy can be evaluated using a modified form of
Cochran’s Q statistic with respect to differences in MVMR estimates
across the set of instruments. In this case, observed heterogeneity is
indicative of a violation of the exclusion restriction assumption in MR
(validity), which can result in biased effect estimates.</p>
<p>Importantly, weak instruments can increase the false positive rate
for pleiotropy detection, as heterogeneity in effect estimates due to
weak instrument bias is conflated with heterogeneity as a result of
pleiotropic bias. As a correction it is possible to estimate
heterogeneity from pleiotropy through Q-statistic minimisation.</p>
<p>The function <code><a href="../reference/pleiotropy_mvmr.html">pleiotropy_mvmr()</a></code> can be used to test for
heterogeneity, requiring the same arguments as the
<code><a href="../reference/strength_mvmr.html">strength_mvmr()</a></code>; <code>r_input</code> and
<code>gencov</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pres</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pleiotropy_mvmr.html">pleiotropy_mvmr</a></span><span class="op">(</span>r_input <span class="op">=</span> <span class="va">F.data</span>, gencov <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in pleiotropy_mvmr(r_input = F.data, gencov = 0): Covariance between</span></span>
<span><span class="co">#&gt; effect of genetic variants on each exposure not specified. Fixing covariance at</span></span>
<span><span class="co">#&gt; 0.</span></span>
<span><span class="co">#&gt; Q-Statistic for instrument validity:</span></span>
<span><span class="co">#&gt; 683.0807 on 141 DF , p-value: 4.880403e-72</span></span></code></pre></div>
<p>And with the example covariance matrices from Step 3:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pres</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pleiotropy_mvmr.html">pleiotropy_mvmr</a></span><span class="op">(</span>r_input <span class="op">=</span> <span class="va">F.data</span>, gencov <span class="op">=</span> <span class="va">Xcovmat</span><span class="op">)</span></span>
<span><span class="co">#&gt; Q-Statistic for instrument validity:</span></span>
<span><span class="co">#&gt; 682.843 on 141 DF , p-value: 5.36533e-72</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-5-estimate-causal-effects">Step 5: Estimate causal effects<a class="anchor" aria-label="anchor" href="#step-5-estimate-causal-effects"></a>
</h2>
<p>Two MVMR estimation methods are provided in the <code>MVMR</code>
package. The first method fits an inverse variance weighted (IVW) MVMR
model, providing estimates of the direct effect of each exposure upon
the outcome. This is performed using the <code><a href="../reference/ivw_mvmr.html">ivw_mvmr()</a></code>
function as shown below:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ivw_mvmr.html">ivw_mvmr</a></span><span class="op">(</span>r_input <span class="op">=</span> <span class="va">F.data</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in ivw_mvmr(r_input = F.data): Covariance between effect of genetic</span></span>
<span><span class="co">#&gt; variants on each exposure not specified. Fixing covariance at 0.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Multivariable MR</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;               Estimate Std. Error   t value  Pr(&gt;|t|)</span></span>
<span><span class="co">#&gt; exposure1 -0.021845061 0.01417255 -1.541364 0.1254538</span></span>
<span><span class="co">#&gt; exposure2  0.003735249 0.01033779  0.361320 0.7183973</span></span>
<span><span class="co">#&gt; exposure3  0.025572042 0.01601913  1.596344 0.1126351</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residual standard error: 2.197 on 142 degrees of freedom</span></span></code></pre></div>
<p>In this case, the effect estimates are interpreted as the direct
effects of LDL-C (exposure 1), HDL-C (exposure 2), and Trg (exposure 3)
on SBP. Estimates are not robust to weak instruments of pleiotropic
bias, and therefore rely upon the underlying MVMR assumptions being
satisfied.</p>
</div>
<div class="section level2">
<h2 id="step-6-robust-causal-effect-estimation-">Step 6: Robust causal effect estimation.<a class="anchor" aria-label="anchor" href="#step-6-robust-causal-effect-estimation-"></a>
</h2>
<p>Where the MVMR assumptions are potentially violated, specifically
where instruments are weak or exhibit pleiotropy, it is possible to
obtain more robust estimates through Q-statistic minimisation. This can
be performed using the <code><a href="../reference/qhet_mvmr.html">qhet_mvmr()</a></code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qhet_mvmr.html">qhet_mvmr</a></span><span class="op">(</span><span class="va">F.data</span>, <span class="va">mvmrcormatrix</span>, CI <span class="op">=</span> <span class="cn">FALSE</span>, iterations <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in qhet_mvmr(F.data, mvmrcormatrix, CI = FALSE, iterations = 100):</span></span>
<span><span class="co">#&gt; qhet_mvmr() is currently undergoing development.</span></span>
<span><span class="va">res1</span></span>
<span><span class="co">#&gt;            Effect Estimates</span></span>
<span><span class="co">#&gt; Exposure 1    -0.0264865644</span></span>
<span><span class="co">#&gt; Exposure 2     0.0094372624</span></span>
<span><span class="co">#&gt; Exposure 3     0.0002575009</span></span></code></pre></div>
<p>It is important to highlight that the phenotypic covariance matrix is
used as an input, and not the set of estimated covariance matrices which
previously formed the <code>gencov</code> argument. It should also be
noted that as the number of exposure and instruments increases,
estimation using <code><a href="../reference/qhet_mvmr.html">qhet_mvmr()</a></code> may prove difficult, owing to
the substantial amount of computing power required. This can be
initially relaxed by not computing 95% confidence intervals as
above.</p>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-sanderson-statsmed-2021" class="csl-entry">
Sanderson, E., W. Spiller, and J. Bowden. 2021. <span>“<span class="nocase">Testing and correcting for weak and pleiotropic
instruments in two-sample multivariable Mendelian
randomization</span>.”</span> <em>Statistics in Medicine</em> 40 (25):
5434–52. <a href="https://doi.org/10.1002/sim.9133" class="external-link">https://doi.org/10.1002/sim.9133</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Wes Spiller, Jack Bowden, Eleanor Sanderson, Tom Palmer.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
