<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>mvmr (legacy) — mvmr • MVMR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="mvmr (legacy) — mvmr" />
<meta property="og:description" content="Note: This function is from the old version of the MVMR package and will be replaced in the future: The gencov argument should be set to zero when using mvmr()." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MVMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MVMR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/WSpiller/MVMR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>mvmr (legacy)</h1>
    <small class="dont-index">Source: <a href='https://github.com/WSpiller/MVMR/blob/master/R/mvmr.R'><code>R/mvmr.R</code></a></small>
    <div class="hidden name"><code>mvmr.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Note: This function is from the old version of the MVMR package and will be replaced in the future: The gencov argument should be set to zero when using <code>mvmr()</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>mvmr</span><span class='op'>(</span><span class='va'>r_input</span>, <span class='va'>gencov</span>, <span class='va'>weights</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>r_input</th>
      <td><p>A formatted data frame using the <code><a href='format_mvmr.html'>format_mvmr</a></code> function or an object of class <code>MRMVInput</code> from <code><a href='https://rdrr.io/pkg/MendelianRandomization/man/mr_mvinput.html'>MendelianRandomization::mr_mvinput</a></code></p></td>
    </tr>
    <tr>
      <th>gencov</th>
      <td><p>Calculating heterogeneity statistics requires the covariance between the effect of the genetic variants on each exposure to be known. This can either be estimated from individual level data, be assumed to be zero, or fixed at zero using non-overlapping samples of each exposure GWAS. A value of <code>0</code> is used by default.</p></td>
    </tr>
    <tr>
      <th>weights</th>
      <td><p>A value specifying the inverse variance weights used to calculate IVW estimate and Cochran's Q statistic. Currently only first order weights are available (<code>1</code>).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>"MVMRIVW"</code> containing the following components:</p><dl>
<dt><code>summary</code></dt><dd><p>A summary of the MVMR regression model, including estimated coefficients, standard errors, t-statistics, p-values, and heterogeneity statistics.</p></dd>
<dt><code>coef</code></dt><dd><p>The estimated coefficients, their standard errors, t-statistics, and corresponding (two-sided) p-values.</p></dd>
<dt><code>Q_strength</code></dt><dd><p>A data frame displaying modified Cochran's Q statistics for assessing instrument strength with respect to each exposure. The Q-statistic increases proportionally with instrument strength, and analogous to univariate MR analyses, a value equal to or greater than 10 can be used as a minimum threshold for instrument strength. Note that for these statistics it is not informative to evaluate p-values.</p></dd>
<dt><code>Q_valid</code></dt><dd><p>A modified form of Cochran's Q statistic measuring heterogeneity in causal effect estimates obtained using each genetic variant. Observed heterogeneity is indicative of a violation of the exclusion restriction assumption in MR (validity), which can result in biased effect estimates.</p></dd>
<dt><code>p_valid</code></dt><dd><p>A p-value corresponding to the heterogeneity measure for instrument validity (<code>Q_valid</code>)</p></dd>

</dl>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Fits an IVW multivariable Mendelian randomization model using first order weights. The function returns an object of class <code>"MVMRIVW"</code>, containing regression estimates, estimated heterogeneity as a measure of instrument strength (<code>Q_strength</code>), and estimated heterogeneity as a measure of instrument validity (<code>Q_valid</code>).</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Sanderson, E., et al., An examination of multivariable Mendelian randomization in the single-sample and two-sample summary data settings. International Journal of Epidemiology, 2019, 48, 3, 713-727. doi: <a href='https://doi.org/10.1093/ije/dyy262'>10.1093/ije/dyy262</a></p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Wes Spiller; Eleanor Sanderson; Jack Bowden.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Example using format_mvmr formatted data</span>
<span class='va'>r_input</span> <span class='op'>&lt;-</span> <span class='fu'><a href='format_mvmr.html'>format_mvmr</a></span><span class='op'>(</span>
    BXGs <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>[</span>,<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"LDL_beta"</span>,<span class='st'>"HDL_beta"</span><span class='op'>)</span><span class='op'>]</span>,
    BYG <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>$</span><span class='va'>SBP_beta</span>,
    seBXGs <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>[</span>,<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"LDL_se"</span>,<span class='st'>"HDL_se"</span><span class='op'>)</span><span class='op'>]</span>,
    seBYG <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>$</span><span class='va'>SBP_se</span>,
    RSID <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>$</span><span class='va'>SNP</span><span class='op'>)</span>
<span class='fu'>mvmr</span><span class='op'>(</span><span class='va'>r_input</span>, <span class='fl'>0</span>, <span class='fl'>1</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; Multivariable MR
#&gt; 
#&gt;               Estimate Std. Error    t value  Pr(&gt;|t|)
#&gt; exposure1 -0.031003996 0.01302925 -2.3795686 0.0186526
#&gt; exposure2  0.006039167 0.01029181  0.5867933 0.5582678
#&gt; 
#&gt; Residual standard error: 2.209 on 143 degrees of freedom
#&gt; 
#&gt; F-statistic: 3.27 on 2 and 143 DF, p-value: 0.0408
#&gt; 
#&gt; ------------------------------
#&gt; Q-Statistics for instrument strength:
#&gt; 
#&gt;   exposure1 exposure2
#&gt; Q  9739.922  10401.49
#&gt; 
#&gt; ------------------------------
#&gt; Q-Statistic for instrument validity:
#&gt; 
#&gt; 695.5924 on 142 DF , p-value: 7.338e-74</div><div class='input'>
<span class='co'># Example using MRMVInput formatted data from the MendelianRandomization package</span>
<span class='va'>bx</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='va'>rawdat_mvmr</span><span class='op'>[</span>,<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"LDL_beta"</span>, <span class='st'>"HDL_beta"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>bxse</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='va'>rawdat_mvmr</span><span class='op'>[</span>,<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"LDL_se"</span>, <span class='st'>"HDL_se"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>dat</span> <span class='op'>&lt;-</span> <span class='fu'>MendelianRandomization</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/MendelianRandomization/man/mr_mvinput.html'>mr_mvinput</a></span><span class='op'>(</span>bx <span class='op'>=</span> <span class='va'>bx</span>,
                                          bxse <span class='op'>=</span> <span class='va'>bxse</span>,
                                          by <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>$</span><span class='va'>SBP_beta</span>,
                                          byse <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>$</span><span class='va'>SBP_se</span>,
                                          snps <span class='op'>=</span> <span class='va'>rawdat_mvmr</span><span class='op'>$</span><span class='va'>SNP</span><span class='op'>)</span>
<span class='fu'>mvmr</span><span class='op'>(</span>r_input <span class='op'>=</span> <span class='va'>r_input</span>, gencov <span class='op'>=</span> <span class='fl'>0</span>, weights <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; Multivariable MR
#&gt; 
#&gt;               Estimate Std. Error    t value  Pr(&gt;|t|)
#&gt; exposure1 -0.031003996 0.01302925 -2.3795686 0.0186526
#&gt; exposure2  0.006039167 0.01029181  0.5867933 0.5582678
#&gt; 
#&gt; Residual standard error: 2.209 on 143 degrees of freedom
#&gt; 
#&gt; F-statistic: 3.27 on 2 and 143 DF, p-value: 0.0408
#&gt; 
#&gt; ------------------------------
#&gt; Q-Statistics for instrument strength:
#&gt; 
#&gt;   exposure1 exposure2
#&gt; Q  9739.922  10401.49
#&gt; 
#&gt; ------------------------------
#&gt; Q-Statistic for instrument validity:
#&gt; 
#&gt; 695.5924 on 142 DF , p-value: 7.338e-74</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Wes Spiller, Jack Bowden, Eleanor Sanderson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


