<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Note: This function is from the old version of the MVMR package and will be replaced in the future: The gencov argument should be set to zero when using mvmr()."><title>mvmr (legacy) — mvmr • MVMR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="mvmr (legacy) — mvmr"><meta property="og:description" content="Note: This function is from the old version of the MVMR package and will be replaced in the future: The gencov argument should be set to zero when using mvmr()."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MVMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MVMR.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/WSpiller/MVMR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>mvmr (legacy)</h1>
      <small class="dont-index">Source: <a href="https://github.com/WSpiller/MVMR/blob/HEAD/R/mvmr.R" class="external-link"><code>R/mvmr.R</code></a></small>
      <div class="d-none name"><code>mvmr.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Note: This function is from the old version of the MVMR package and will be replaced in the future: The gencov argument should be set to zero when using <code>mvmr()</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mvmr</span><span class="op">(</span><span class="va">r_input</span>, <span class="va">gencov</span>, <span class="va">weights</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>r_input</dt>
<dd><p>A formatted data frame using the <code><a href="format_mvmr.html">format_mvmr</a></code> function or an object of class <code>MRMVInput</code> from <code><a href="https://rdrr.io/pkg/MendelianRandomization/man/mr_mvinput.html" class="external-link">MendelianRandomization::mr_mvinput</a></code></p></dd>


<dt>gencov</dt>
<dd><p>Calculating heterogeneity statistics requires the covariance between the effect of the genetic variants on each exposure to be known. This can either be estimated from individual level data, be assumed to be zero, or fixed at zero using non-overlapping samples of each exposure GWAS. A value of <code>0</code> is used by default.</p></dd>


<dt>weights</dt>
<dd><p>A value specifying the inverse variance weights used to calculate IVW estimate and Cochran's Q statistic. Currently only first order weights are available (<code>1</code>).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>An object of class <code>"MVMRIVW"</code> containing the following components:</p><dl><dt><code>summary</code></dt>
<dd><p>A summary of the MVMR regression model, including estimated coefficients, standard errors, t-statistics, p-values, and heterogeneity statistics.</p></dd>

<dt><code>coef</code></dt>
<dd><p>The estimated coefficients, their standard errors, t-statistics, and corresponding (two-sided) p-values.</p></dd>

<dt><code>Q_strength</code></dt>
<dd><p>A data frame displaying modified Cochran's Q statistics for assessing instrument strength with respect to each exposure. The Q-statistic increases proportionally with instrument strength, and analogous to univariate MR analyses, a value equal to or greater than 10 can be used as a minimum threshold for instrument strength. Note that for these statistics it is not informative to evaluate p-values.</p></dd>

<dt><code>Q_valid</code></dt>
<dd><p>A modified form of Cochran's Q statistic measuring heterogeneity in causal effect estimates obtained using each genetic variant. Observed heterogeneity is indicative of a violation of the exclusion restriction assumption in MR (validity), which can result in biased effect estimates.</p></dd>

<dt><code>p_valid</code></dt>
<dd><p>A p-value corresponding to the heterogeneity measure for instrument validity (<code>Q_valid</code>)</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Fits an IVW multivariable Mendelian randomization model using first order weights. The function returns an object of class <code>"MVMRIVW"</code>, containing regression estimates, estimated heterogeneity as a measure of instrument strength (<code>Q_strength</code>), and estimated heterogeneity as a measure of instrument validity (<code>Q_valid</code>).</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Sanderson, E., et al., An examination of multivariable Mendelian randomization in the single-sample and two-sample summary data settings. International Journal of Epidemiology, 2019, 48, 3, 713-727. <a href="https://doi.org/10.1093/ije/dyy262" class="external-link">doi:10.1093/ije/dyy262</a></p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Wes Spiller; Eleanor Sanderson; Jack Bowden.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example using format_mvmr formatted data</span></span></span>
<span class="r-in"><span><span class="va">r_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="format_mvmr.html">format_mvmr</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    BXGs <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LDL_beta"</span>,<span class="st">"HDL_beta"</span><span class="op">)</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>    BYG <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">$</span><span class="va">SBP_beta</span>,</span></span>
<span class="r-in"><span>    seBXGs <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LDL_se"</span>,<span class="st">"HDL_se"</span><span class="op">)</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>    seBYG <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">$</span><span class="va">SBP_se</span>,</span></span>
<span class="r-in"><span>    RSID <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">$</span><span class="va">SNP</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">mvmr</span><span class="op">(</span><span class="va">r_input</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Multivariable MR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               Estimate Std. Error    t value  Pr(&gt;|t|)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> exposure1 -0.031003996 0.01302925 -2.3795686 0.0186526</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> exposure2  0.006039167 0.01029181  0.5867933 0.5582678</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Residual standard error: 2.209 on 143 degrees of freedom</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> F-statistic: 3.27 on 2 and 143 DF, p-value: 0.0408</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------------------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q-Statistics for instrument strength:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   exposure1 exposure2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q  9739.922  10401.49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------------------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q-Statistic for instrument validity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 695.5924 on 142 DF , p-value: 7.338e-74</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example using MRMVInput formatted data from the MendelianRandomization package</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"MendelianRandomization"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">bx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">rawdat_mvmr</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LDL_beta"</span>, <span class="st">"HDL_beta"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">bxse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">rawdat_mvmr</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LDL_se"</span>, <span class="st">"HDL_se"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">MendelianRandomization</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MendelianRandomization/man/mr_mvinput.html" class="external-link">mr_mvinput</a></span><span class="op">(</span>bx <span class="op">=</span> <span class="va">bx</span>,</span></span>
<span class="r-in"><span>                                          bxse <span class="op">=</span> <span class="va">bxse</span>,</span></span>
<span class="r-in"><span>                                          by <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">$</span><span class="va">SBP_beta</span>,</span></span>
<span class="r-in"><span>                                          byse <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">$</span><span class="va">SBP_se</span>,</span></span>
<span class="r-in"><span>                                          snps <span class="op">=</span> <span class="va">rawdat_mvmr</span><span class="op">$</span><span class="va">SNP</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">mvmr</span><span class="op">(</span>r_input <span class="op">=</span> <span class="va">r_input</span>, gencov <span class="op">=</span> <span class="fl">0</span>, weights <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Multivariable MR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               Estimate Std. Error    t value  Pr(&gt;|t|)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> exposure1 -0.031003996 0.01302925 -2.3795686 0.0186526</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> exposure2  0.006039167 0.01029181  0.5867933 0.5582678</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Residual standard error: 2.209 on 143 degrees of freedom</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> F-statistic: 3.27 on 2 and 143 DF, p-value: 0.0408</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------------------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q-Statistics for instrument strength:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   exposure1 exposure2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q  9739.922  10401.49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------------------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q-Statistic for instrument validity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 695.5924 on 142 DF , p-value: 7.338e-74</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Wes Spiller, Jack Bowden, Eleanor Sanderson.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

